File: Week_01/G20190343020308/remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        length = len(nums);
        if length == 0:
            return 0
        try:
            for i in range(0,length-1):
                temp = nums[i];
                while temp == nums[i+1]:
                    del nums[i];
                    
            
        except IndexError:
                pass

        return len(nums);

File: Week_01/G20190343020224/RemoveDuplicatesFromSortedArray.java
class RemoveDuplicatesFromSortedArray {

    class Solution {

        //双指针：快慢指针 用带索引的for循环要考虑越界，用foreach不用
        public int removeDuplicates(int[] nums) {
            if (nums.length == 0) {
                return 0;
            }
            int index = 0;
            for (int i: nums) {
                if (i > nums[index]) {
                    nums[++index] = i;
                }
            }
            return index + 1;
        }

    }

}